alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,232 @@ \section{TxInfo Construction}
+      & ~~~~ \times \seqof{\type{P.PubKeyHash}}  \\
+      & ~~~~\text{$\uparrow$ hashes of keys whose signature are required} \\ % = map transKeyHash (Set.toList (reqSignerHashes' tbody)), \type{P.txInfoSignatories}
+      & ~~~~ \times \seqof{(\DataHash \times \Data)}  \\
+      & ~~~~ \text{$\uparrow$ list of pairs of datum hashes and their datum preimages} \\ % = map transDataPair datpairs, \type{P.txInfoData}
+      & ~~~~ \times \type{P.TxId} \\ % =  (transSafeHash (hashAnnotated @(Crypto era) tbody)) \type{P.txInfoId}
+      & ~~~~ \text{$\uparrow$ list of pairs of datum hashes and their datum preimages}
+  \end{align*}
+  \caption{TxInfo and Constituent Types}
+  \label{fig:txinfo-types-two}
+\end{figure*}
+
+\begin{figure*}[htb]
+  \begin{align*}
+    & \fun{transAddr} \in \Addr  \to  \type{P.Address} \\
+    & \fun{transAddr} ~a = \begin{cases}
+      \Nothing & \text{if}~a~\in~\AddrBS \\
+      (ob,~st) & \text{if}~a~=~(\wcard, ob_P, st_P)
+    \end{cases} \\
+    & \text{Address translation}
+    \nextdef
+    & \fun{transValue} \in \Value \to \type{P.Value} \\
+    & \fun{transValue}~ (c,~ mp) = \{~ pid_P\mapsto (aid_P \mapsto q_P) ~\mid~ pid \mapsto (aid \mapsto q) ~\in ~mp ~\} \\
+    & ~~~~\cup \{~\type{P.adaSymbol}\mapsto (\type{P.adaToken} \mapsto c) ~\} \\
+    & \text{Value translation}
+    \nextdef
+    & \fun{transDCert} \in \DCertRegPool \to \type{P.DCertPoolRegister} \\
+    & \fun{transDCert}~\var{pool} = (\fun{poolId}~\var{pool},~\fun{poolVrf}~\var{pool}) \\
+    & \text{Pool registration certificate translation}
+    \nextdef
+    & \fun{txInfoIn} \in \UTxO \to \TxIn \to \type{P.TxInInfo} \\
+    & \fun{txInfoIn}~ utxo~ txin = \begin{cases}
+        (\var{txin}_P, \var{txout}_P) & \text{when $(\var{txin}\mapsto\var{txout})\in\var{utxo}$} \\
+        \Nothing & \text{otherwise}
+      \end{cases} \\
+    & \text{Builds a Plutus realized input}
+    \nextdef
+    & \fun{transVITime} \in \PParams \to \EpochInfo \to \SystemStart \to \ValidityInterval \to \type{P.POSIXTimeRange} \\
+    & \fun{transVITime} ~pp ~ei~ sysS~ (vs,~ vf) = \\
+    &  ~~~~ \begin{cases}
+        \type{P.always} & \text{if } vs = \Nothing ~\wedge~ vf = \Nothing \\
+        \type{P.to}~(\fun{slotToPOSIXTime}~pp~ei~sysS~vf) & \text{if } vs = \Nothing ~\wedge~ vf \neq \Nothing \\
+        \type{P.from}~(\fun{slotToPOSIXTime}~pp~ei~sysS~vs) & \text{if } vs \neq \Nothing ~\wedge~ vf = \Nothing \\
+        (\fun{slotToPOSIXTime}~pp~ei~sysS~vs,~\fun{slotToPOSIXTime}~pp~ei~sysS~vf) & \text{if } vs \neq \Nothing ~\wedge~ vf \neq \Nothing \\
+      \end{cases} \\
+      & ~~~~\where \\
+      & ~~~~~~~~\fun{slotToPOSIXTime}~pp~ei~sysS~vs~=~\type{ResolutionConstant} \\
+    & \text{Translate validity interval to a POSIX time range}
+  \end{align*}
+  \caption{TxInfo Constituent Type Translation Functions}
+  \label{fig:txinfo-translations}
+\end{figure*}
+
+\begin{figure*}[htb]
```

Comment:
There's too much white space around all the `=` signs in this figure (remove `~`)